src/solvers/cholmod.jl

```diff
@@ -151,7 +151,18 @@ macro cholmod_param(kwarg, code)
     end
 end
 
-const COMMONS = Vector{Ref{cholmod_common}}()
+function newcommon(; print = 0)
+    common = finalizer(cholmod_l_finish, Ref(cholmod_common()))
+    result = cholmod_l_start(common)
+    @assert result == TRUE "failed to run `cholmod_l_start`!"
+    common[].print = 0  # no printing from CHOLMOD by default
+    common[].error_handler = @cfunction(error_handler, Cvoid, (Cint, Cstring, Cint, Cstring))
+    return common
+end
+
+function getcommon()
+    return get!(task_local_storage(), :cholmod_common, newcommon())::Ref{cholmod_common}
```

Comment:
Should this not be
```
get!(newcommon, task_local_storage(), :cholmod_common)::Ref{cholmod_common}
```
? As it is now a new object is created for every call.

Reply:
The docs say it's `collection, key, default` right?

Reply:
There's another method where you can give a function/type as a first arg. Note that Fredrick's suggestion gives the type but doesn't evaluate it